src/main/java/uk/ac/ic/wlgitbridge/bridge/repo/WalkOverrideGitRepo.java

```diff
@@ -0,0 +1,90 @@
+package uk.ac.ic.wlgitbridge.bridge.repo;
+
+import org.eclipse.jgit.lib.ObjectId;
+import org.eclipse.jgit.lib.Repository;
+import uk.ac.ic.wlgitbridge.data.filestore.GitDirectoryContents;
+import uk.ac.ic.wlgitbridge.data.filestore.RawDirectory;
+import uk.ac.ic.wlgitbridge.git.exception.GitUserException;
+import uk.ac.ic.wlgitbridge.git.util.RepositoryObjectTreeWalker;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Optional;
+
+/**
+ * We compose because we don't want to deal with the static constructors
+ */
+public class WalkOverrideGitRepo implements ProjectRepo {
+
+    private final GitProjectRepo gitRepo;
+
+    private final Optional<Long> maxFileSize;
+
+    private final Optional<ObjectId> commitId;
+
```

Comment:
This is because the default GitProjectRepo does not have a file size limit or care about file size limits, and we override the behaviour using this class. But maybe we should just make it non-optional with a default size limit, e.g. 50 MB, based on what @jdleesmiller said below.

Reply:
OK, docs have been added to the corresponding PR. I'll assign you to look over once the prev ones have been deployed.